compilers/concrete-compiler/compiler/lib/Conversion/SimulateTFHE/SimulateTFHE.cpp

```diff
@@ -92,6 +92,67 @@ struct NegOpPattern : public mlir::OpConversionPattern<TFHE::NegGLWEOp> {
   }
 };
 
+template <typename AddOp, typename AddOpAdaptor>
+struct AddOpPattern : public mlir::OpConversionPattern<AddOp> {
+
+  AddOpPattern(mlir::MLIRContext *context, mlir::TypeConverter &typeConverter)
+      : mlir::OpConversionPattern<AddOp>(
+            typeConverter, context,
+            mlir::concretelang::DEFAULT_PATTERN_BENEFIT) {}
+
+  ::mlir::LogicalResult
+  matchAndRewrite(AddOp addOp, AddOpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+
+    const std::string funcName = "sim_add_lwe_u64";
+
+    if (insertForwardDeclaration(
+            addOp, rewriter, funcName,
```

Comment:
Could be great to also give the location to be printed in the overflow detection